torch/_inductor/select_algorithm.py

```diff
@@ -366,6 +372,46 @@ def __init__(
         self.template_out: Optional[str] = None
         self.ops_handler: Optional[V.WrapperHandler] = None  # type: ignore[name-defined]
 
+        # The generated code is not dependent on the input nodes names, or symbolic sizes names.
```

Comment:
This is just for the template generation of non-fused templates. It wont be true for templates which undergo fusion. I'm not sure exactly how we're avoiding that today. Could we more explicitly opt out of caching for that case ?

We only attempt to fuse epilogues onto one template of the original set by default, so its not worth caching anyway. But we might attempt more. You can test this with adding a `relu` at the end of an mm and backends=TRITON i mentioned above.

This is the code path for it https://github.com/pytorch/pytorch/blob/5a422150c32f2c861061dc8be5f8d19fb6d80155/torch/_inductor/codegen/simd.py#L1461-L1471

Reply:
We should also skip if subgraphs exist. Or for flexattention / subgraph_bodies..

Reply:
this code path is not on the cache path. i.e: it does not call generate_and_load since it direcly call render
that said we are redundantly recording the state:
here is what I will do to address there.

1) as mentioned in a different comment, I will make this opt in so we control locations that actually uses the caching .
2) I will update PR so that the recording does not happen when not needed.

Reply:
addressed as explained.